_test/test_sqw/test_noisify.m

```diff
@@ -0,0 +1,141 @@
+    test_sqw_file_path = '../test_sqw_file/deterministic_sqw_fake_data_for_testing.sqw';
+    test_sqw_file_full_path = '';
+
+    pixel_data_obj;
+    pix_data_from_file;
+    pix_data_from_faccess;
+    pix_data_small_page;
+    pix_fields = {'u1', 'u2', 'u3', 'dE', 'coordinates', 'q_coordinates', ...
+                  'run_idx', 'detector_idx', 'energy_idx', 'signal', ...
+                  'variance'};
+end
+
+properties (Constant)
+    NUM_BYTES_IN_VALUE = 8;
+    NUM_COLS_IN_PIX_BLOCK = 9;
+end
+
+methods (Access = private)
+
+    function pix_data = get_random_pix_data_(~, rows)
+        data = rand(9, rows);
+        pix_data = PixelData(data);
+    end
+
+end
+
+methods
+
+    function obj = test_noisify(~)
+        obj = obj@TestCase('test_noisify');
+
+        % Swallow any warnings for when pixel page size set too small
+        obj.old_warn_state = warning('OFF', 'PIXELDATA:validate_mem_alloc');
+
+        % Process the file path
+        test_sqw_file = 'c:\boobly6.sqw';%java.io.File('c:\', obj.test_sqw_file_path);
+        obj.test_sqw_file_full_path = 'c:\boobly6.sqw';%char(test_sqw_file.getCanonicalPath());
+
+        % Tests use a number of different input streams:
+        % Construct an object from raw data
+        obj.pixel_data_obj = PixelData(obj.raw_pix_data);
+        % Construct an object from a file
+        obj.pix_data_from_file = PixelData(obj.test_sqw_file_path);
+        % Construct an object from a file accessor
+        f_accessor = sqw_formats_factory.instance().get_loader(obj.test_sqw_file_path);
+        obj.pix_data_from_faccess = PixelData(f_accessor);
+        % Construct an object from file accessor with small page size
+        obj.pix_data_small_page = PixelData(f_accessor, obj.small_page_size_);
+    end
+
+    function delete(obj)
+        warning(obj.old_warn_state);
+    end
+
+    % --- Tests for in-memory operations ---
+    
+    function test_how_noisify_works(obj)
+        % Thinking aloud here:
+        % obj is the test case object
+        % what we want to do is call noisify here on different kinds of
+        % data
+        % step 1 we reduce the page size
+        hc = hor_config();
+        hc.pixel_page_size = 10000;
+        % we set up the test "random number generator" which is actually
+        % a deterministic set of numbers 1:999 repeated. Use factor to make
+        % them in range 0:1
+        factor =1/999;
+        % We make an sqw object with the above page size
+        sqw_obj1 = sqw(obj.test_sqw_file_full_path);
+        % and we noisify it
+        rng(0) % don't need seed as it's not the proper rng but left in in
+        % case we change our mind
+        % instead set the flag to initialize the test "random" sequence
+        global testrand_init;
+        testrand_init = true;
+        % add the noise to object 1
+        
+        addpath('./utils')
+        addpath('../../../Herbert/_test/test_utilities');
+        addpath('../../../Herbert/_test/shared');
```

Comment:
Is it necessary to explicitly add `Herbert/_test/shared` to the path? 

There are other tests (e.g. `Horace\_test\test_tobyfit\test_tobyfit_1`) using functions in the folder that do not include the path update.

Reply:
ed: I've just noticed `test_tobyfit_1` is using a local copy of `seed_rng` and not the one from the `shared` folder so ignore this comment.